alacritty_terminal/src/term/search.rs

```diff
@@ -290,7 +285,11 @@ impl<T> Term<T> {
     ) {
         match direction {
             Direction::Right if cell.flags.contains(Flags::WIDE_CHAR) => {
-                iter.next();
+                // If advances `iter` at the end of column one character is skipped
+                // unintentionally.
+                if iter.point().column < self.columns() - 1 {
+                    iter.next();
+                };
```

Comment:
For the sake of consistency, the conditional should probably be moved into the match arm. Otherwise it's somewhat arbitrary.